crates/typst/src/eval/call.rs

```diff
@@ -32,135 +34,48 @@ impl Eval for ast::FuncCall<'_> {
         }
 
         // Try to evaluate as a call to an associated function or field.
-        let (callee, mut args) = if let ast::Expr::FieldAccess(access) = callee {
-            let target = access.target();
-            let target_span = target.span();
+        let (callee, args) = if let ast::Expr::FieldAccess(access) = callee {
+            let target_expr = access.target();
             let field = access.field();
-            let field_span = field.span();
+            let mut args = args.eval(vm)?.spanned(span);
 
+            // Evaluate the target expression.
             let target = if is_mutating_method(&field) {
-                let mut args = args.eval(vm)?.spanned(span);
-                let target = target.access(vm)?;
-
-                // Only arrays and dictionaries have mutable methods.
-                if matches!(target, Value::Array(_) | Value::Dict(_)) {
-                    args.span = span;
```

Comment:
Paging @laurmaedje : I think this `args.span = span;` was at odds with the original `.spanned(span)`, right? It does seem that `.spanned(span)` only changes the span if there wasn't already one, while `.span = span` will always change it. I think `.spanned(span)` is enough here, since `.eval` creates an `Args` with a detached span, so this should be ok to be removed, correct? (Just wanted to make sure we're doing the right thing here)

Reply:
When investigating this code, I did think the extra `.span = span` is no need here, because `.spanned()` already set it.

If I am wrong, please correct me. :smile: